cmake/testing.cmake

```diff
@@ -12,6 +12,8 @@ set(DOCTEST_NO_INSTALL True)
 
 FetchContent_MakeAvailable(doctest)
```

Comment:
This is not required, but the fecth content requires something like this https://github.com/a4z/zq/blob/b66ce8048751c29556f2caf360eff7e89d3c6cb0/CMakeLists.txt#L28

and it would be nice if the whole thing stays optional, if possible is already?

Reply:
Nice! This should also fix the pipeline build